test/controllers/deploys_controller_test.rb

```diff
@@ -216,6 +215,74 @@
     unauthorized :delete, :destroy, project_id: :foo, id: 1
   end
 
+  as_a_viewer do
+    before do
+      Deploy.delete_all()
+      Job.delete_all()
+      job1 = Job.create({project_id: Project.first.id, command:'cap staging deploy', status:'failed', user_id: admin.id})
+      Deploy.create({ stage_id: Stage.find_by_production(true).id, reference: 'reference', job_id: job1.id})
+      job2 = Job.create({project_id: Project.first.id, command:'cap staging deploy', status:'running', user_id: admin.id})
+      Deploy.create({ stage_id: Stage.find_by_production(true).id, reference: 'reference', job_id: job2.id})
+      success1 = Job.create({project_id: Project.first.id, command:'cap staging deploy', status:'succeeded', user_id: admin.id})
+      Deploy.create({ stage_id: Stage.find_by_production(true).id, reference: 'reference', job_id: success1.id})
+      success2 = Job.create({project_id: Project.first.id, command:'cap staging deploy', status:'succeeded', user_id: admin.id})
+      Deploy.create({ stage_id: Stage.find_by_production(false).id, reference: 'reference', job_id: success2.id})
+    end
+
+    after do
+      Deploy.delete_all()
+    end
+
+    describe "a GET to :search" do
+      it "returns a 200" do
+        get :search, format: "json"
+        assert_response :ok
+      end
+
+      it "should return no results" do
```

Comment:
can get rid of all the `shoulds` 

`it "returns no results for unfound deployer"`